[pytorch][dynamo_compile] Log inductor config to dynamo_compile

ppanchalia opened the pull request:
Summary:
Scrubbed inductor config logging to dynamo_compile as json:str.

Scrub RE: `r'((^TYPE_CHECKING$)|(.*_progress$)|(.*TESTING.*)|(.*(rocm|halide).*)|(^trace\..*)|(^_))'`to save some space.

Test Plan:
Staging logger: https://fburl.com/data/ltkt08zm

P1679697917

{F1958428018}

Differential Revision: D65806399




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

ppanchalia commented:
@pytorchbot label "topic: not user facing"

masnesral commented on torch/_dynamo/utils.py:
```diff
@@ -906,6 +908,20 @@ def add_compilation_metrics_to_chromium(c: Dict[str, Any]) -> None:
         dynamo_config=c["dynamo_config"],
     )
 
+def scrubbed_inductor_config_for_logging() -> Dict[str, Any]:
```
nit: can you prefix this function with underscore to (hopefully) communicate that it's "private" to this file?

masnesral commented on torch/_dynamo/utils.py:
```diff
@@ -906,6 +908,20 @@ def add_compilation_metrics_to_chromium(c: Dict[str, Any]) -> None:
+    configs_to_scrub_re = r'((^TYPE_CHECKING$)|(.*_progress$)|(.*TESTING.*)|(.*(rocm|halide).*)|(^trace\..*)|(^_))'
+    keys_to_scrub = set()
+    inductor_config_copy = inductor_config.get_config_copy()
+    for key, val in inductor_config_copy.items():
```
nit: Since we have to iterate over all the items anyway, I feel like the original request to make this a comprehension just looks nicer. Something like this?

```
scrubbed_config = {key, val for key, val in inductor_config.items() if isinstance(val,...) and not re.search(...)}
```

ppanchalia replied:
So inductor_config(torch._inductor.config) is not really a dict that we could do a `.items()` on, the `get_config_copy()` converts it to as a copy of type `Dict[str, Any]`. If we do comprehensions, we'll be creating another copy inevitably. I'd like to avoid that :)

masnesral commented on torch/_dynamo/utils.py:
```diff
@@ -906,6 +908,20 @@ def add_compilation_metrics_to_chromium(c: Dict[str, Any]) -> None:
+    for key, val in inductor_config_copy.items():
+        if re.search(configs_to_scrub_re, key):
+            keys_to_scrub.add(key)
+        if not isinstance(val, str|bool|int|float):
```
Is this valid syntax?

ppanchalia replied:
It is, but I'll change it to typing.get_args(Union[str,int,bool,float]) so it is more readable

ppanchalia replied:
Meh, fixed it

masnesral commented:
I think you'll need to update one of the new tests I wrote. I'd just do the same thing as I did with the dynamo_config field and clear it out: https://github.com/pytorch/pytorch/blob/main/test/dynamo/test_utils.py#L197

masnesral reviewed (changes requested):
requesting changes for test failure

masnesral commented on test/dynamo/test_utils.py:
```diff
@@ -326,6 +329,83 @@ def test_dynamo_timed(self, mock_time, mock_time_ns):
+            self.a = a
+            self.b = b
+
+    """
```
nit: The python style seems to be to always put these docs under the first line of the method decl instead of above

masnesral commented on test/dynamo/test_utils.py:
```diff
@@ -326,6 +329,83 @@ def test_dynamo_timed(self, mock_time, mock_time_ns):
+        try:
+            inductor_config_json = utils._scrubbed_inductor_config_for_logging()
+        except Exception as e:
+            self.fail(
```
I don't think you need to explicitly catch the exception. Test will fail if there's an exception

masnesral commented on test/dynamo/test_utils.py:
```diff
@@ -326,6 +329,83 @@ def test_dynamo_timed(self, mock_time, mock_time_ns):
+            self.fail(
+                f"utils._scrubbed_inductor_config_for_logging raised an exception:{e}"
+            )
+        assert isinstance(inductor_config_json, str)
```
self.assertTrue

masnesral commented on torch/_dynamo/utils.py:
```diff
@@ -921,6 +962,8 @@ def us_to_ms(field):
         return metric // 1000 if metric is not None else None
 
+    # Log inductor_config to compilation metrics
+    metrics["inductor_config"] = _scrubbed_inductor_config_for_logging()
```
Instead of mutating the param, can you instead just add this to the `legacy_metrics` dict? We can rename it to `common_metrics` maybe. Something like:

```
common_metrics = {
    "inductor_config": _scrubbed_inductor_config_for_logging(),
    ... any future common metrics would go here ...
    # TODO: Temporary; populate legacy fields from their replacements.
    # Remove when we decide we can really deprecate them.
    ... all the existing "legacy" fields go here ...
 }
```

ppanchalia committed:
[pytorch][dynamo_compile] Log inductor config to dynamo_compile (#140790)

Summary:
Scrubbed inductor config logging to dynamo_compile as json:str.

Scrub RE: `r'((^TYPE_CHECKING$)|(.*_progress$)|(.*TESTING.*)|(.*(rocm|halide).*)|(^trace\..*)|(^_))'`to save some space.


Test Plan:
Staging logger: https://fburl.com/data/ltkt08zm

P1679697917

{F1958428018}





cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

Reviewed By: masnesral

Differential Revision: D65806399

Pulled By: ppanchalia

ppanchalia commented:
@pytorchbot merge
